### ISSUE TITLE
Documentation of event handlers

### ISSUE THREAD

**Issue by b747fp** [fullcalendar/fullcalendar#5449]
The Docs for FullCalendar are a joke.
I click on Handlers to get a list of handlers. there's no list of handlers... just 1 small example, which i am not looking for...
https://fullcalendar.io/docs/handlers

... where can i get some real documentation for v4?

**Comment by acerix** [comment]
You could reference this list of all props/events if that's what you are looking to:

https://github.com/fullcalendar/fullcalendar-vue/blob/v4/src/fullcalendar-options.js

**Comment by dclinton182** [comment]
I'm also struggling to find examples of all events/handlers I can use with the calendar which led me to this page.

Can the full list (and examples) be added to this page (https://fullcalendar.io/docs/handlers) in the docs as it seems like the logical place for it to be in my opinion?

**Comment by lucianobosco** [comment]
What about V5 handlers? I've expended 2 days and I can't figure out which events are emitted when next/prev buttons are clicked by using the timeGridPlugin on Vue component.
I need to grab events data from backend when user navigates, but no luck so far

**Comment by devsware** [comment]
Quite frustrating the lack of documentation about handlers for V5.

In some situations, as a temporary workaround you could wrap your code inside a Promise and do something like this (the example initialize a new calendar and only after it has been rendered print a message on console):

```
let myCalendar = null;

let initCalendar = new Promise((resolve,reject) => {
    myCalendar = new FullCalendar.Calendar(document.getElementById("my-calendar"), {
        initialView: 'dayGridMonth'
    });
    
    myCalendar.render();
    resolve(myCalendar);
});

initCalendar.then(calendar => {
    console.info('It seems the calendar has been rendered');
}).catch(err) {
    console.log(err);
}
```

**Comment by JFGagnon** [comment]
> What about V5 handlers? I've expended 2 days and I can't figure out which events are emitted when next/prev buttons are clicked by using the timeGridPlugin on Vue component.
> I need to grab events data from backend when user navigates, but no luck so far

Any luck finding the correct event? For such a big library, the documentation is a joke.

**Comment by bastien70** [comment]
Something really needs to be done, I need the list of possible handlers on Fullcalendar v5 ...

**Comment by dclinton182** [comment]
@acerix - Based on the recent comments I think this one may need reopening unless there is already another task related to improving the documentation for handlers?

**Comment by max-docly** [comment]
I totally agree, more documentation is needed. How are we expected to use the calendar if we don't know all the possible props/handlers that exist?

**Comment by rijkvanzanten** [comment]
@arshaw Why was this issue closed? Having to scout through the source code to find these is/was less than ideal, and it's not obvious how we can utilize them, and if we should even rely on them in the first place 🤔 

Would you accept a PR to the docs to add a reference list of all handlers? 

---

This is what I was able to find by searching through the `fullcalendar` repo:

These are the v5 handlers as per [the `CALENDAR_LISTENER_REFINERS` type](https://github.com/fullcalendar/fullcalendar/blob/f6e69de679267c700084bafa8a3e1a017c1c3387/packages/common/src/options.ts#L287) (omitting the internal ones):

- `datesSet`
- `eventAdd`
- `eventChange`
- `eventClick`
- `eventMouseEnter`
- `eventMouseLeave`
- `eventRemove`
- `eventsSet`
- `loading`
- `select`
- `unselect`
- `windowResize`

The following are emitted, but not documented in the TypeScript definitions (they're not internal though, as those are seemingly prefixed with `_` 🤔 )

- `dateClick`
- `dragend`
- `dragmove`
- `dragstart`
- `drop`
- `eventDragStart`
- `eventDragStop`
- `eventDrop`
- `eventLeave`
- `eventReceive`
- `eventResize`
- `eventResizeStart`
- `eventResizeStop`
- `hitupdate`
- `pointerdown`
- `pointermove`
- `pointerup`

With the following only being available in premium:

- `resourcesSet`
- `resourceAdd`
- `resourceChange`
- `resourceRemove`

And the following only in premium "scrollgrid":

- `scrollStart`
- `scrollEnd`
- `scroll`

**Comment by bizunas** [comment]
"scrollgrid" `scrollStart`, `scrollEnd` and `scroll` are not being emmited to calendar api emitter, but to local one which would not fire your handlers.

**Comment by thesnowgoose** [comment]
I had to create my own events for the nex/prev buttons click and for the view buttons click

`  React.useEffect(() => {
    const [dayButton] = document.getElementsByClassName('fc-dayGridDay-button');
    const [weekButton] = document.getElementsByClassName('fc-dayGridWeek-button');
    const [monthButton] = document.getElementsByClassName('fc-dayGridMonth-button');
    const [prevButton] = document.getElementsByClassName('fc-prev-button');
    const [nextButton] = document.getElementsByClassName('fc-next-button');

    const onClickPrevNext = calculateDate => () => myHandler(calculateDate);
    const onSelectView = view => myViewHandler(view);

    const changeMonthView = () => onSelectView('dayGridMonth');
    const changeWeekView = () => onSelectView('dayGridWeek');
    const changeDayView = () => onSelectView('dayGridDay');
    const onClickPrev = onClickPrevNext(decreaseDate);
    const onClickNext = onClickPrevNext(increaseDate);

    monthButton.addEventListener('click', changeMonthView);
    weekButton.addEventListener('click', changeWeekView);
    dayButton.addEventListener('click', changeDayView);
    nextButton.addEventListener('click', onClickNext);
    prevButton.addEventListener('click', onClickPrev);

    return () => {
      monthButton.removeEventListener('click', changeMonthView);
      weekButton.removeEventListener('click', changeWeekView);
      dayButton.removeEventListener('click', changeDayView);
      nextButton.removeEventListener('click', onClickNext);
      prevButton.removeEventListener('click', onClickPrev);
    };
  }, []);`

**Comment by bramirez96** [comment]
Over a year later and documentation still has not been updated, what's the status on this?

**Comment by lucianobosco** [comment]
> Over a year later and documentation still has not been updated, what's the status on this? My company wants to use the paid version of the product but doesn't feel comfortable spending the money on a project with so little documentation.

I won't do that seeing that last update was 9 months ago and there is no response in regards to docs. Also pending PRs since 2017...
